utilities/pipelines/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -27,62 +87,78 @@ function Invoke-ResourceRemoval {
         [string] $Type
     )
 
-    switch ($type) {
+    Remove-UnhandledResourceLock -ResourceId $ResourceId
```

Comment:
I wonder - shouldn't this logic simply be part of the `Microsoft.Authorization/locks` switch case? 

Unless I'm missing something, I think the logic works like this
- A list of resources comes in and is sorted by the order they have to be removed in
- Lock comes first
- In then run's into the below switch case and starts removing it
- Then it continues with the rest.

You anyways added a 'retry' logic to that switch-case. Did you experience it still not being enough? Right now it looks like some fancy dupication :D

Reply:
ok, fair enough. Then how about that - we invoke your inner function (which should proably be in its own file) **also** inside the 'lock' switch case. Both invoke the `Remove-AzResourceLock` function, so it would be cool if we could somehow combine them to a commmon logic.

I think the one real difference is that in one case we get the lock from the resource (`Get-AzResourceLock`), while the other directly uses the resource ID inserted in the function. So maybe it would be enough to have some sort of case handling in the beginning and have the rest (including the removal & loop) use the same logic.

So we would still have 2 invocations, but at least only one piece of logic.

Reply:
Restructured so that the resource lock removal uses same logic.